vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/treegrid/TreeGrid.java

```diff
@@ -796,6 +796,10 @@ public void expandRecursively(Stream<T> items, int depth) {
     public void expandRecursively(Collection<T> items, int depth) {
         getDataCommunicator()
                 .expand(getItemsWithChildrenRecursively(items, depth));
+        // Call with client roundtrip as expansion takes a while.
+        // This ensures that event is fired after exapnsion is complete.
+        getElement().executeJs("return 1;").then(result -> fireEvent(
+                new ExpandEvent<T, TreeGrid<T>>(this, false, items)));
```

Comment:
The JavaDoc for this method says `This method will <i>not</i> fire events for expanded nodes.`

Reply:
I changed the issue type to enhancement